build_tools/benchmarks/common/benchmark_config.py

```diff
@@ -108,12 +108,23 @@ def real_path_or_none(
           capture_tarball=args.capture_tarball.resolve(),
           capture_tmp_dir=per_commit_tmp_dir / CAPTURES_REL_PATH)
 
-    build_dir = args.build_dir.resolve()
-    if args.run_config is None:
-      # TODO(#11076): Remove legacy path.
-      root_benchmark_dir = build_dir / BENCHMARK_SUITE_REL_PATH
+    if args.e2e_test_artifacts_dir is not None:
```

Comment:
This sort of validation seems like it would be better suited to the argument parser itself

Reply:
Done. Created a subclass of ArgumentParser to do validation.

Reply:
Oh I just meant you could do it in the `parse_args` method instead of the benchmark config